modules/api/src/main/java/org/apache/ignite/lang/IgniteException.java

```diff
@@ -252,4 +254,30 @@ public int errorCode() {
     public UUID traceId() {
         return traceId;
     }
+
+
+    /**
+     * Wraps another exception in IgniteException, extracting {@link #traceId} and {@link #code} when the specified exception
+     * or one of its causes is an IgniteException itself.
+     *
+     * @param e Internal exception.
+     * @return Public exception.
+     */
+    public static IgniteException wrap(Throwable e) {
+        e = ExceptionUtils.unwrapCause(e);
+
+        if (e instanceof IgniteException) {
```

Comment:
It seems to me that we need to add support for `IgniteCheckedException` as well.

Reply:
Good catch, done.